src/Symfony/Component/Mailer/Envelope.php

```diff
@@ -44,6 +45,9 @@ public static function create(RawMessage $message): self
 
     public function setSender(Address $sender): void
     {
+        if (preg_match('/.*[^\x00-\x7F]+.*@/', $sender->getAddress())) {
+            throw new AddressEncoderException(sprintf('Non-ASCII characters not supported in local-part os "%s".', $sender->getAddress()));
```

Comment:
@nicolas-grekas not sure if this exception makes sense semantically here? We are not really encoding anything yet, or?

Reply:
`InvalidArgumentException` then?

Reply:
Yes I think it makes more sense :+1: